batch/batch/server/database.py

```diff
@@ -122,6 +122,35 @@ def __init__(self, db):
     async def has_record(self, id):
         return await super().has_record({'id': id})
 
+    async def find_records(self, user, complete=None, success=None):
+        values = []
+        sql = "select batch.* from {self.name} as batch"
+        joins = []
+        wheres = []
+        if complete or success:
+            joins += "inner join {self._db.batch_jobs.name} as bj using (batch_id)"
+            joins += "inner join {self._db.jobs.name} as job using (job_id)"
+            if complete is not None:
+                values += "Complete"
+                if complete:
+                    wheres += "job.state = %s"
+                else:
+                    wheres += "job.state != %s"
```

Comment:
Can you add a test to make sure this is working? I'm not sure `using` is correct since they have different id names in the two tables.